src/utils/units.cpp

```diff
@@ -0,0 +1,100 @@
+//========================================================================================
+// (C) (or copyright) 2024. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+#include "units.hpp"
+
+namespace ArtemisUtils {
+
+Units::Units(ParameterInput *pin) {
+  unit_system_ = pin->GetOrAddString("artemis/units", "type", "code");
+  if (unit_system_ == "code") {
+    length_ = 1.;
+    time_ = 1.;
+    mass_ = 1.;
+    G_ = 1.;
+    kb_ = 1.;
```

Comment:
I don't know if code units is `kb=1` or `Rgas=1`. I tend to use `Rgas=1` (and I think the default `cv` assumes that right now).

Reply:
I'd personally prefer never to think about $R$ 😬  My brain is too hardwired to:

rho k T / (mu mp) == gm1 rho e

Reply:
Default units here is `kb = mp = 1`; I think `kb` makes more sense in this case because singularity-eos interacts with the specific heat capacity `<c_v> = erg / (g * K) -> c_v = k_b / (gam - 1) mu m_p`.

You are right that I need to actually calculate `c_v` correctly as part of this PR